.github/workflows/gsprocessing_workflow.yml

```
@@ -0,0 +1,120 @@
+# Graphstorm CI for GSProcessing Only
+# Test execution order - [lint] -> [pytest]
+name: gsprocessing_workflow
+
+on:
+  push:
+    branches:
+      - main
+      - ci_dev
+    paths:
+      - '.github/workflow_scripts/gsprocessing.sh'
+      - 'graphstorm-processing/*'
+      - '.github/workflows/gsprocessing_workflow.yml'
+    workflow_dispatch:
+  pull_request_target:
+    paths:
+      - '.github/workflow_scripts/gsprocessing.sh'
+      - 'graphstorm-processing/*'
+      - '.github/workflows/gsprocessing_workflow.yml'
+
+permissions:
+  id-token: write
+  pull-requests: write
+  contents: read
+
+defaults:
+  run:
+    shell: bash
+
+# All test scripts split in individual .sh files and moved to .github/workflow scripts
+env:
+  COMMAND-LINT: chmod +x ./lint_check.sh && ./lint_check.sh
+  COMMAND-GSProcessing: chmod +x ./gsprocessing.sh && ./gsprocessing.sh
```

The commands can just be:

```suggestion
  COMMAND-LINT: bash ./lint_check.sh
  COMMAND-GSProcessing-PYTEST: bash ./gsprocessing-pytest.sh
```

---

Note that the above recommendation changes the name of the command/env var, as well as the name of script for pytest.